pkg/reconcile/reconcile.go

```diff
@@ -91,3 +93,31 @@ var _ Reconciler = Func(nil)
 
 // Reconcile implements Reconciler.
 func (r Func) Reconcile(o Request) (Result, error) { return r(o) }
+
+// PanicHandler can be called with defer in order to recover from panic in reconcile logic.
+// Additional handlers can also be passed to manage the go routines. The error passed to the
+// PanicHandler is set to the value of panic if not nil.
+// If any resource triggers a panic, the reconciler will recover and modify the raised panic
+// to the normal reconciler error.
```

Comment:
This must be called with `defer`.
Also some suggestions on clarifying how and why the error is set.
```suggestion
// PanicHandler is meant to be called with defer in the Reconcile function in order to recover from a panic. 
// Additional handlers can be passed for more custom logic to handle the panic e.g shutting down go routines. 
// PanicHandler will set the error, err, as the panic message and the stack trace.
// This can be used to set the Reconciler's return error to safely retry when recovering from a panic. 
```